src/main/java/seedu/address/logic/parser/ChangeCommandParser.java

```diff
@@ -0,0 +1,71 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_EXPIRED_DATE_TIME;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DURATION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TIME;
+import static seedu.address.model.appointment.Appointment.CREATION_OFFSET_MINUTES;
+
+import java.time.Duration;
+import java.time.LocalTime;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.DateTimeUtil;
+import seedu.address.logic.commands.ChangeCommand;
+import seedu.address.logic.commands.ChangeCommand.EditAppointmentDescriptor;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.appointment.Date;
+import seedu.address.model.appointment.Time;
+
+
+/**
+ * Parses input arguments and creates a new ChangeCommand object.
+ */
+public class ChangeCommandParser implements Parser<ChangeCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the ChangeCommand
+     * and returns an ChangeCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format.
+     */
+    public ChangeCommand parse(String arg) throws ParseException {
+        requireNonNull(arg);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(arg, PREFIX_DATE, PREFIX_TIME, PREFIX_DURATION);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ChangeCommand.MESSAGE_USAGE), pe);
+        }
+
+        if (argMultimap.getValue(PREFIX_DATE).isEmpty()) {
```

Comment:
Should we check in the other way? 
Coz in this case, the command can be successful with even just one of the prefix content present. Hence `isPresent()` should be used instead of `isEmpty()`
I think you can refer to the `EditCommandParser` class

Reply:
Yes I agree with Wanlin, `isPresent()` would be better. The user can change to a specific date, time or duration without changing the rest.